File: ComponentImpl/src/main/java/com/xiaojinzi/component/impl/FragmentNavigator.java
package com.xiaojinzi.component.impl;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.xiaojinzi.component.impl.fragment.FragmentManager;

/**
 * 一个为 {@link Fragment} 设计的导航器
 */
public class FragmentNavigator {

    @NonNull
    protected String fragmentFlag;

    @NonNull
    protected Bundle bundle;

    public FragmentNavigator(@NonNull String fragmentFlag) {
        this.fragmentFlag = fragmentFlag;
    }

    public FragmentNavigator bundle(@NonNull Bundle bundle) {
        this.bundle = bundle;
        return this;
    }

    @Nullable
    public Fragment navigate() {
        return FragmentManager.get(fragmentFlag, bundle);
    }

}

File: ComponentImpl/src/main/java/com/xiaojinzi/component/impl/FragmentNavigator.kt
package com.xiaojinzi.component.impl

import com.xiaojinzi.component.impl.fragment.FragmentManager.get
import androidx.fragment.app.Fragment
import com.xiaojinzi.component.support.IBundleBuilder
import com.xiaojinzi.component.support.IBundleBuilderImpl

/**
 * 一个为 [Fragment] 设计的导航器
 */
class FragmentNavigator(private val fragmentFlag: String): IBundleBuilder<FragmentNavigator> by IBundleBuilderImpl() {

    fun navigate(): Fragment? {
        return get(fragmentFlag, bundle)
    }

}